Sources/Testing/Events/Recorder/Event.AdvancedConsoleOutputRecorder.swift

```diff
@@ -0,0 +1,826 @@
+      summaryParts.append("\(symbol) \(context.symbolCounts["skip", default: 0])")
+    }
+    
+    if context.symbolCounts["warning", default: 0] > 0 {
+      let symbol = Event.Symbol.warning.stringValue(options: options.base)
+      summaryParts.append("\(symbol) \(context.symbolCounts["warning", default: 0])")
+    }
+    
+    if context.symbolCounts["difference", default: 0] > 0 {
+      let symbol = Event.Symbol.difference.stringValue(options: options.base)
+      summaryParts.append("\(symbol) \(context.symbolCounts["difference", default: 0])")
+    }
+    
+    if context.symbolCounts["details", default: 0] > 0 {
+      let symbol = Event.Symbol.details.stringValue(options: options.base)
+      summaryParts.append("\(symbol) \(context.symbolCounts["details", default: 0])")
+    }
+    
+    if context.symbolCounts["attachment", default: 0] > 0 {
+      let symbol = Event.Symbol.attachment.stringValue(options: options.base)
+      summaryParts.append("\(symbol) \(context.symbolCounts["attachment", default: 0])")
+    }
+    
+    if !summaryParts.isEmpty {
+      return summaryParts.joined(separator: " ") + "\n"
+    }
+    
+    return ""
+  }
+  
+  private func getTestCount(from eventContext: borrowing Event.Context) -> Int {
+
+    return 0
+  }
+  
+  private func isPassedStatus(_ status: HierarchyNode.NodeStatus) -> Bool {
+    switch status {
+    case .passed:
+      return true
+    default:
+      return false
+    }
+  }
+  
+  // MARK: - Signal Handling
+  
+  private static func setupSignalHandlers() {
+    Task { @MainActor in
+      // Only set up the signal handler once
+      guard Self._signalHandler == nil else { return }
+      
+      Self._hasActiveRecorder = true
+      
+      #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+      let signalType = SIGINFO
+      #else
+      let signalType = SIGUSR1
+      #endif
+      
+      signal(signalType, SIG_IGN) // Ignore default handler
+      
+      let signalSource = DispatchSource.makeSignalSource(signal: signalType, queue: .main)
+      signalSource.setEventHandler {
+        Self.printGlobalOnDemandStatus()
+      }
+      signalSource.resume()
+      Self._signalHandler = signalSource
+    }
+  }
+  
+  @MainActor private static func printGlobalOnDemandStatus() {
+    guard _hasActiveRecorder else {
+      try? FileHandle.stderr.write("\nNo active test recorder found.\n")
+      return
+    }
+    
+    try? FileHandle.stderr.write("\nAdvanced console output recorder is active.\n")
+  }
+  
+  private static func cleanupSignalHandlers() {
+    Task { @MainActor in
+      Self._hasActiveRecorder = false
+    }
+  }
+}
+
+// MARK: - Extensions
+
+extension Int {
+  /// Get a human-readable description of this instance as a count of some item.
+  ///
+  /// - Parameters:
+  ///   - item: The item being counted.
+  ///
+  /// - Returns: A human-readable description such as "1 test" or "2 tests".
+  fileprivate func counting(_ item: String) -> String {
```

Comment:
I think on a previous PR I suggested factoring this utility out so it could be shared with the standard console output recorder